docs/rules/no-js-extension.md

```diff
@@ -36,6 +43,55 @@ define(function (require) {
     var foo = require('foo');
     /* ... */
 });
+
+require(['pluginname!foo'], function (foo) {
```

Comment:
Based on how the diff was laid out I mistakenly thought your addition was under the " The following patterns are considered warnings:" section. Sorry about that!